hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -44,6 +44,13 @@
       .sinceVersion("0.7.0")
       .withDocumentation("Enable the internal metadata table which serves table metadata like level file listings");
 
+  // Enable syncing the Metadata Table
+  public static final ConfigProperty<Boolean> METADATA_SYNC_ENABLE_PROP = ConfigProperty
```

Comment:
here what's the difference between config `hoodie.metadata.enable`, can we reuse the config?

Reply:
that just controls whether hudi writer should populate the metadata table at all. This PR is introducing a writer level config which can be used to say prevent compaction/clustering etc from syncing to metadata table.

Reply:
this config and that config `hoodie.metadata.enable` confused me, from the users end, when setting `hoodie.metadata.enable` to true, we expect hudi should create and sync to metadata table, here we introduce another similar config and would make confusion.

Reply:
@leesf the current problem is that metadata table is not safe when written from multiple pipelines together. So we need to keep it enabled in all pipelines but only have one pipeline write to it. 

Two alternate suggestion:
1. Change the name of this config :
The writing-to-metadata-table is being called the "sync" here. We can change it to read-only mode?
hoodie.metadata.readonly=true?

2. Introduce metadata table mode instead of enable disable:
hoodie.metadata.mode=[disabled,readwrite,readonly].    readwrite and readonly imply enabled=true and users only provide the single config mode.

I also do not want another config to confuse users. The other option is to enable multi-writer mode for metadata table. I have that implemented as part of another PR (metadata table next version): https://github.com/apache/hudi/pull/3426/commits/f417390aed56c05dbaedeac6ecf47294996bb591

Reply:
I think enable multi-writer mode for metadata table is a better solution. And I go through the codebase of metadata sync, in `preWrite`, syncing metadata is in transaction(and should be ok in multi writing pipeline?) and I see the difference between setting the `hoodie.metadata.enable` to true and `hoodie.metadata.sync.enable` to true is creating `SparkHoodieBackedTableMetadataWriter`,  while in `postWrite`, it is not in transaction(should not ok in multi writing pipeline). Would you please correct me in some other aspect I missed? @prashantwason

Reply:
You are right. 

I feel the correct way would be have the transaction within the Metadata Table itself so we dont have to worry about which codepath it is called from.

Reply:
given this is last min for RC cut, I suggest we keep this for now. the checks I added should alleviate the case for sigle writer, with async cleaning, compaction, clustering already. 

We can pour our energy into sync design and may be even remve this config.